src/api/api.js

```diff
@@ -19,441 +53,322 @@ export const getMessageConversations = ({
     assignedToMeFilter,
     markedForFollowUpFilter,
     unreadFilter,
+    currentUser,
 }) => {
-    const filters = [`messageType:eq:${messageType}`]
-    typeof status !== 'undefined' && filters.push(`status:eq:${status}`)
-    typeof priority !== 'undefined' && filters.push(`priority:eq:${priority}`)
-    markedForFollowUpFilter && filters.push('followUp:eq:true')
-    unreadFilter && filters.push('read:eq:false')
-
-    return getD2Instance()
-        .then(instance => {
-            assignedToMeFilter &&
-                filters.push(`assignee.id:eq:${instance.currentUser.id}`)
-
-            return instance.Api.getApi().get(
-                `messageConversations?pageSize=${pageSize}&page=${page}${
-                    messageFilter !== '' && messageFilter !== undefined
-                        ? `&queryString=${messageFilter}`
-                        : ''
-                }`,
-                {
-                    fields: [initialMessageConversationFields],
-                    order,
-                    filter: filters,
-                }
-            )
-        })
-        .then(result => ({
-            messageConversations: result.messageConversations,
-            pager: result.pager,
-        }))
-        .catch(error => {
-            throw error
-        })
+    const filter = [`messageType:eq:${messageType}`]
+
+    if (status) {
+        filter.push(`status:eq:${status}`)
+    }
+    if (priority) {
+        filter.push(`priority:eq:${priority}`)
+    }
+    if (markedForFollowUpFilter) {
+        filter.push('followUp:eq:true')
+    }
+    if (unreadFilter) {
+        filter.push('read:eq:false')
+    }
+    if (assignedToMeFilter) {
+        filter.push(`assignee.id:eq:${currentUser.id}`)
+    }
+
+    const query = {
+        resource: 'messageConversations',
+        params: {
+            filter,
+            pageSize,
+            page,
+            fields: initialMessageConversationFields,
+            order,
+        },
+    }
+
+    if (messageFilter) {
```

Comment:
I think I'd expect this to be created before the query assignment (as say `const queryString = messageFilter`) so that you don't have to mutate the query. Seems more fitting with the rest of the logic to me.

Reply:
I thought that might result in a `queryString=undefined` in the GET, but I just checked the relevant app-runtime code, and [this turns out not to be the case](https://github.com/dhis2/app-runtime/blob/master/services/data/src/links/RestAPILink/queryToResourcePath.ts#L41-L46). I'll adjust this.

Reply:
Addressed in 6750a171b2ef38a68d180085da54e26a02e5cd8c